src/libstore/gc.cc

```diff
@@ -330,18 +335,184 @@ Roots LocalStore::findRoots()
 }
 
 
+static void readProcLink(int procFd, const char * file, StringSet & paths)
```

Comment:
Better to just pass a full path here rather than a file descriptor (e.g. don't use `*at()` system calls).